Close #67.  Update events index to table

matthewrpacker opened the pull request:
@GSmes & @ryanflach 
- There is odd behavior with the search functionality and the events index table.  There is a blank table with table rows for every event displayed above the table that correctly displays every event table row.  Only the blank table seems to be filtering with the search functionality.  Please take a look when you both have a chance.  Thanks.

matthewrpacker committed:
Merge branch 'responsive_hamburger_navbar'

matthewrpacker committed:
Merge branch 'develop' of github.com:ryanflach/the_pivot

matthewrpacker committed:
[WIP]Update events index to table

ryanflach commented on app/views/shared/_event_list.html.erb:
```diff
@@ -1,24 +1,47 @@
+            <th>Location</th>
+            <th>Date</th>
+            <th>Price</th>
+            <th>Purchase</th>
```
@matthewrpacker consider adding class 'text-center' to purchase - purely aesthetic but I think it looks better with the 'Add to Cart' button.

ryanflach commented on app/views/shared/_event_list.html.erb:
```diff
@@ -1,24 +1,47 @@
+        </thead>
+        <tbody>
+          <% @events.each do |event| %>
+            <div class='event' data-all="<%= event.title %> <%= event.venue_name %> <%= event.category.title %> <%= event.supporting_act %>">
```
@matthewrpacker This is the main culprit for this not working. Remove this outer `<div>`.

ryanflach commented on app/views/shared/_event_list.html.erb:
```diff
@@ -1,24 +1,47 @@
+        <tbody>
+          <% @events.each do |event| %>
+            <div class='event' data-all="<%= event.title %> <%= event.venue_name %> <%= event.category.title %> <%= event.supporting_act %>">
+              <div class ="well col-md-12 event-in-list" id="event-<%= event.id %>">
```
@matthewrpacker Don't need this line with the table.

ryanflach commented on app/views/shared/_event_list.html.erb:
```diff
@@ -1,24 +1,47 @@
+          <% @events.each do |event| %>
+            <div class='event' data-all="<%= event.title %> <%= event.venue_name %> <%= event.category.title %> <%= event.supporting_act %>">
+              <div class ="well col-md-12 event-in-list" id="event-<%= event.id %>">
+                <tr id="event-<%= event.id %>">
```
@matthewrpacker This is where you want to set the `data-all` attribute. Your `<tr>` line should look like:
`<tr class="event" id="event-<%= event.id %>" data-all="<%= event.title %> <%= event.venue_name %> <%= event.category.title %> <%= event.supporting_act %> <%= event.venue_location %> <%= event.event_date %>">`

Note: I added additional data so that we could also search by location and date.

ryanflach commented on app/views/shared/_event_list.html.erb:
```diff
@@ -1,24 +1,47 @@
+                    <a id="popoverData" href="<%= event_path(event.venue, event) %>" data-content="<%= event.supporting_act %>" rel="popover" data-placement="bottom" data-original-title="Supporting Act" data-trigger="hover">
+                      <%= event.title %>
+                    </a>
+                  </td>
```
Don't need this `<td>` anymore.

ryanflach commented on app/views/shared/_event_list.html.erb:
```diff
@@ -1,24 +1,47 @@
+                  <td><%= event.event_date %></td>
+                  <td><%= number_to_currency(event.price) %></td>
+                  <td>
+                    <%= content_tag :div do %>
```
@matthewrpacker This no longer needs to be a div (we only need the `button_to`.

ryanflach commented:
@matthewrpacker I made a number of line comments. Here is a sample of working code for `_event_list.html.erb`:

```
<div id="events_list">
  <div class="well col-md-10 padded">
    <div class="table-responsive col-md-8 fit-width">
      <table class="table">
        <thead>
          <tr>
            <th>Event</th>
            <th>Venue</th>
            <th>Category</th>
            <th>Location</th>
            <th>Date</th>
            <th>Price</th>
            <th class='text-center'>Purchase</th>
          </tr>
        </thead>
        <tbody>
          <% @events.each do |event| %>
            <tr class="event" id="event-<%= event.id %>" data-all="<%= event.title %> <%= event.venue_name %> <%= event.category.title %> <%= event.supporting_act %> <%= event.venue_location %> <%= event.event_date %>">
              <td><%= event.title %></td>
              <td><%= event.venue_name %></td>
              <td><%= event.category.title %></td>
              <td><%= event.venue_location %></td>
              <td><%= event.event_date %></td>
              <td><%= number_to_currency(event.price) %></td>
              <td>
                <%= button_to "#{btn_label}", cart_events_path(event_id: event.id), class: "btn btn-#{btn_class}", disabled: status %>
              </td>
            </tr>
          <% end %>
        </tbody>
      </table>
    </div>
  </div>
</div>
```

Only thing the above does not have is an edit button `if current_admin?`. You'd need to add that as a `<td>`.

matthewrpacker committed:
Update events index to table
- Make event name a link
- Make event.venue_name a link
- Make event.category a link
- Add supporting act as a column
